src/st/st0/30030.c

```diff
@@ -471,22 +469,23 @@ void PrologueScroll(void) {
     }
 }
 
+extern Dialogue2 g_Dialogue;
```

Comment:
Might be worth a comment here to make it super obvious that g_Dialogue is being used weirdly in this function. Otherwise, this is a good solution to a weird problem. Nice to avoid polluting the main Dialogue struct.